src/c_kzg_4844.c

```diff
@@ -698,90 +698,52 @@ static C_KZG_RET blob_to_polynomial(Polynomial *p, const Blob *blob) {
     return C_KZG_OK;
 }
 
-/* Forward function definition */
-static void compute_powers(fr_t *out, fr_t *x, uint64_t n);
+/* Input size to the Fiat-Shamir challenge computation */
+#define CHALLENGE_INPUT_SIZE 32 + BYTES_PER_BLOB + 48
 
 /**
- * Return the Fiat-Shamir challenges required by the rest of the protocol.
+ * Return the Fiat-Shamir challenge required to verify `blob` and `commitment`.
  *
  * @remark This function should compute challenges even if `n==0`.
  *
  * @param[out] eval_challenge_out The evaluation challenge
- * @param[out] r_powers_out       The powers of r, where r is a randomly
- *                                generated scalar
- * @param[in]  polys              The array of polynomials
- * @param[in]  comms              The array of commitments
+ * @param[in]  blob               A blob
+ * @param[in]  commitment         A commitment
  * @param[in]  n                  The number of polynomials and commitments
  */
-static C_KZG_RET compute_challenges(
-    fr_t *eval_challenge_out,
-    fr_t *r_powers_out,
-    const Polynomial *polys,
-    const g1_t *comms,
-    uint64_t n
+static void compute_challenge(
+    fr_t *eval_challenge_out, const Blob *blob, const g1_t *commitment
 ) {
-    C_KZG_RET ret;
-    size_t i;
-    uint64_t j;
-    uint8_t *bytes = NULL;
-
-    // len(FIAT_SHAMIR_PROTOCOL_DOMAIN) + 8 + 8 + n blobs + n commitments
-    size_t input_size = 32 + (n * BYTES_PER_BLOB) + (n * 48);
-    ret = c_kzg_malloc((void **)&bytes, input_size);
-    if (ret != C_KZG_OK) goto out;
+    Bytes32 eval_challenge;
+    uint8_t bytes[CHALLENGE_INPUT_SIZE];
 
     /* Pointer tracking `bytes` for writing on top of it */
     uint8_t *offset = bytes;
 
     /* Copy domain separator */
     memcpy(offset, FIAT_SHAMIR_PROTOCOL_DOMAIN, 16);
     offset += 16;
-    bytes_from_uint64(offset, FIELD_ELEMENTS_PER_BLOB);
+    bytes_from_uint64(
+        offset, 0
```

Comment:
shouldn't the polynomial degree (`FIELD_ELEMENTS_PER_BLOB`) go here instead of `0`?

Reply:
oh I see you need extra zero padding...